From cc8f3ee26a65f442ed99e195b50535cd0734199a Mon Sep 17 00:00:00 2001
From: Hu Kejun <william.hu@rock-chips.com>
Date: Mon, 4 Mar 2019 08:25:55 +0800
Subject: [PATCH] media: rockchip: isp1: change readx_poll_timeout to
 readx_poll_timeout_atomic

rkisp1_isp_stop() is called in rkisp1_isp_isr() now,
so change readx_poll_timeout in rkisp1_isp_stop() to readx_poll_timeout_atomic

Change-Id: If4cb8628e38d2c774f30d7200b0804f22d4ccc4c
Signed-off-by: Hu Kejun <william.hu@rock-chips.com>
---
 drivers/media/platform/rockchip/isp1/rkisp1.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/drivers/media/platform/rockchip/isp1/rkisp1.c b/drivers/media/platform/rockchip/isp1/rkisp1.c
index 8366442d6f13..6dfd405ab2ee 100644
--- a/drivers/media/platform/rockchip/isp1/rkisp1.c
+++ b/drivers/media/platform/rockchip/isp1/rkisp1.c
@@ -751,8 +751,8 @@ static int rkisp1_isp_stop(struct rkisp1_device *dev)
 	val = readl(base + CIF_ISP_CTRL);
 	writel(val | CIF_ISP_CTRL_ISP_CFG_UPD, base + CIF_ISP_CTRL);
 
-	readx_poll_timeout(readl, base + CIF_ISP_RIS,
-			   val, val & CIF_ISP_OFF, 20, 100);
+	readx_poll_timeout_atomic(readl, base + CIF_ISP_RIS,
+				  val, val & CIF_ISP_OFF, 20, 100);
 	v4l2_dbg(1, rkisp1_debug, &dev->v4l2_dev,
 		"streaming(MP:%d, SP:%d), MI_CTRL:%x, ISP_CTRL:%x, MIPI_CTRL:%x\n",
 		 dev->stream[RKISP1_STREAM_SP].streaming,
-- 
2.35.3

